dev-qt/prison/prison-5.38.recipe

```diff
@@ -0,0 +1,49 @@
+SUMMARY="A barcode api"
+DESCRIPTION="prison has a prison::AbstractBarcode, which is the base class  \
+for the actual barcode generators, currently prison::QRCodeBarcode and \
+prison::DataMatrixBarcode is the two implemented barcode generators."
+HOMEPAGE="https://api.kde.org/frameworks/prison/html/index.html"
+COPYRIGHT="2010-2017 KDE Foundations"
+LICENSE="GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://download.kde.org/stable/frameworks/5.38/prison-5.38.0.tar.xz"
+SOURCE_DIR="prison-5.38.0"
+CHECKSUM_SHA256="95bc65c9164b8ed8cace08be7dfdf12cc9a398a42e2b4c66852faaa1cd1f325f"
+ARCHITECTURES="x86 x86_64"
+
+PROVIDES="
+	Prison = $portVersion
+	lib:libKF5Prison.so = $portVersion
+	lib:libkf5Prison.so.5 = $portVersion
+	"
+REQUIRES="
+	haiku
+	libdmtx
+	qrencode
+	qt5
+	"
+
+BUILD_REQUIRES="
+	libdmtx_devel
+	qt5_devel
+	haiku_devel
+	qrencode_devel
+	extra_cmake_modules
+	"
+BUILD_PREREQUIRES="
+	cmake
+	cmd:gcc
+	cmd:make
+	"
+
+BUILD()
+{
+	export ECM_DIR=/boot/system/data/cmake/Modules/
+	cmake -DCMAKE_INSTALL_PREFIX:PATH=$libDir/cmake/ .
```

Comment:
I just peeked to another KDE's product recipe:
 https://github.com/haikuports/haikuports/blob/master/kde-frameworks/kcrash/kcrash-5.39.0.recipe#L58

https://github.com/haikuports/haikuports/blob/master/kde-frameworks/extra-cmake-modules/extra_cmake_modules-5.39.0.recipe

And i think it should be
```
cmake . \
        -DCMAKE_INSTALL_PREFIX:PATH=$libDir/cmake \
        -DECM_DIR=$dataDir/cmake/Modules/ECM/cmake 
```
Not pretty sure if we can put `$cmakeDirArgs` since i can't find it in wiki, but most KDE's recipe used that.

Reply:
Thanks @raefaldhia for your message!
I've just tried and got this error:
```
waiting for build package prison-5.41.0-1 to be activated
Building ...
-- Could NOT find ECM (missing: ECM_DIR)
-- 
 * ECM (required version >= 5.41.0), Extra CMake Modules., <https://projects.kde.org/projects/kdesupport/extra-cmake-modules>

CMake Error at /boot/system/data/cmake/Modules/FeatureSummary.cmake:459 (message):
  feature_summary() Error: REQUIRED package(s) are missing, aborting CMake
  run.
Call Stack (most recent call first):
  CMakeLists.txt:10 (feature_summary)


-- Configuring incomplete, errors occurred!
See also "/sources/prison-5.41.0/CMakeFiles/CMakeOutput.log".
```
The version of extra_cmake_modules I have is 5.39.0-1. And it looks like we can only build a given version of prison if we have at least the same version (or greater) for extra_cmake_modules.
I'll try to build prison 5.39.0.

Reply:
I managed to build prison 5.39.0 (on x86_64) thanks to your help! Thanks @raefaldhia !